src/index.ts

```diff
@@ -1314,109 +1314,42 @@ export abstract class PinejsClientCore<
 		) => this)(cloneParams, cloneBackendParams);
 	}
 
-	public async get<
-		TResource extends StringKeyOf<Model>,
-		TParams extends Params<Model[TResource]> & {
-			options: {
-				$count: NonNullable<ODataOptions<Model[TResource]['Read']>['$count']>;
-			};
-		},
-	>(
-		params: {
-			resource: TResource;
-		} & TParams,
-	): Promise<number>;
 	public async get<
 		TResource extends StringKeyOf<Model>,
 		TParams extends Params<Model[TResource]> & {
 			id: NonNullable<Params<Model[TResource]>['id']>;
-			options: NonNullable<
-				Params<Model[TResource]>['options'] &
-					(
-						| {
-								$select: NonNullable<
-									NonNullable<Params<Model[TResource]>['options']>['$select']
-								>;
-						  }
-						| {
-								$expand: NonNullable<
-									NonNullable<Params<Model[TResource]>['options']>['$expand']
-								>;
-						  }
-					)
-			>;
 		},
 	>(
 		params: {
 			resource: TResource;
 		} & TParams,
 	): Promise<
 		| NoInfer<
-				PickDeferred<
-					Model[TResource]['Read'],
-					SelectPropsOf<
-						Model[TResource]['Read'],
-						NonNullable<TParams['options']>
-					>
-				> &
-					ExpandToResponse<
+				Extract<
+					OptionsToResponse<
 						Model[TResource]['Read'],
 						NonNullable<TParams['options']>
-					>
+					>,
+					any[]
```

Comment:
I am still sllightly reluctant that picking all arrays here will mess the typings up (altough I do not have a counter example).

Do you think that replacing this `any[]` with `Array<Mode[TResource]['Read']>` would work?

Reply:
I hit some issues when using `Array<Model[TResource]['Read']>`. Presumably because it does not fully match as what we're looking at can be a subset